### PR TITLE
LibRegex: Milk some more performance

### PR DESCRIPTION
**Pull request by alimpfard**
TLDR, this is about a 20% performance increase in our regex test suite, and a whopping 1.19s time save in test262.

### CODE REVIEW

File: Userland/Libraries/LibRegex/RegexMatch.h [hunk 1]
```diff
@@ -225,21 +225,34 @@ class RegexStringView {
 
     size_t length() const
     {
+        if (m_cached_length.has_value())
```
Comment (trflynn89) [inline Userland/Libraries/LibRegex/RegexMatch.h:233]: huh, I'm surprised this was an improvement. The 2 views that are expensive to compute code point length for (UTF-8 and UTF-16) already cache it; and getting code unit length is a constant thing.

Comment (awesomekling) [reply Userland/Libraries/LibRegex/RegexMatch.h:233]: One thing that *is* expensive today is that "length in code points" is not cached across regex executions. Many test262 tests run regexes on the same UTF-16 string repeatedly, but we make a new RegexStringView each time, which then has to re-measure the string.

Comment (alimpfard) [reply Userland/Libraries/LibRegex/RegexMatch.h:233]: 🤔 I recall it being a fairly small improvement, I wouldn't mind removing this and figuring out how to fix the thing AK mentioned

Comment (alimpfard) [reply Userland/Libraries/LibRegex/RegexMatch.h:233]: @awesomekling: the length of the input string is actually not _that_ useful (it's only used if matching in /m mode), there are many intermediate substrings created, and they're really only used once over the lifetime of the regex object.

Comment (alimpfard) [reply Userland/Libraries/LibRegex/RegexMatch.h]: I've removed the patch in question, it just avoided a function call so its effect was very minimal regardless.

**Pull request merged**